upstairs/src/downstairs.rs

```diff
@@ -3185,6 +3187,7 @@ impl Downstairs {
             ds_id,
             client_id,
             responses,
+            vec![], // XXX this could lead to false matches; does it matter?
```

Comment:
I'm surprised this works!

Reply:
Yup, this takes post-decryption data, so it's not actually checking the hashes (other than seeing that each of the 3x reads returns the same thing).  It's a little awkward, but this is a test-only function, so I think we can let it slide.

(I also want to remove `process_ds_completion` entirely at some point; it's only still around because I didn't want to port a zillion unit tests while also refactoring the Upstairs)

Reply:
> so it's not actually checking the hashes (other than seeing that each of the 3x reads returns the same thing)

Yeah, this is why I was surprised, but I guess that makes sense: all three reads did return the same thing (`vec![]`). Is it worth a check to see that _some_ hash was returned?

Reply:
I think it makes sense to assert that the number of hashes == number of blocks in `process_io_completion`; if that's not the case, then it's definitely a logic error in the offloaded decryption!

Implemented in 9d9f15fe498344b81778ec72bdbcea8dc8f13f03, with an explanatory comment 😄